sdk/identity/azure-identity/azure/identity/_credentials/default.py

```diff
@@ -124,6 +123,10 @@ def get_token(self, *scopes, **kwargs):
           `message` attribute listing each authentication attempt and its error message.
         """
         if self._successful_credential:
-            return self._successful_credential.get_token(*scopes, **kwargs)
+            token = self._successful_credential.get_token(*scopes, **kwargs)
```

Comment:
This call must succeed, right? 

Then maybe we can just do 

```python
_LOGGER.info(
    "%s acquired a token from %s", self.__class__.__name__, self._successful_credential.__class__.__name__
 )
return self._successful_credential.get_token(*scopes, **kwargs)
```
?

Reply:
This call could fail. Maybe the credential requests a new token and something goes wrong.